packages/dx-react-chart-demos/src/demo-sources/export/material-ui/demo.jsx

```diff
@@ -0,0 +1,297 @@
+import React, { useState } from 'react';
+import Paper from '@material-ui/core/Paper';
+import {
+  ArgumentAxis,
+  ValueAxis,
+  Chart,
+  Legend,
+  BarSeries,
+  Title,
+} from '@devexpress/dx-react-chart-material-ui';
+import { Plugin, Template, TemplatePlaceholder } from '@devexpress/dx-react-core';
+import domtoimage from 'dom-to-image';
+import JsPDF from 'jspdf';
+import IconButton from '@material-ui/core/IconButton';
+import Menu from '@material-ui/core/Menu';
+import MenuItem from '@material-ui/core/MenuItem';
+import MoreVertIcon from '@material-ui/icons/MoreVert';
+import { scaleBand } from '@devexpress/dx-chart-core';
+import { ArgumentScale, Stack, Animation } from '@devexpress/dx-react-chart';
+import { withStyles } from '@material-ui/core/styles';
+import Typography from '@material-ui/core/Typography';
+
+import { gaming as data } from '../../../demo-data/data-vizualization';
+
+const options = [
+  'Print',
+  'PNG',
+  'JPEG',
+  'PDF',
+];
+
+const styles = () => ({
+  button: {
+    width: '50px',
+    height: '50px',
+  },
+});
+
+const addKeyframe = (name, def) => {
+  if (typeof document === 'undefined') {
+    return;
+  }
+  const head = document.getElementsByTagName('head')[0];
+  let style = Array.from(head.getElementsByTagName('style'))
+    .find(node => node.dataset[Symbol('animation')]);
+  if (!style) {
+    style = document.createElement('style');
+    style.type = 'text/css';
+    style.dataset[Symbol('animation')] = true;
+    head.appendChild(style);
+  }
+  const content = style.textContent;
+  if (!content.includes(name)) {
+    style.textContent += `\n@keyframes ${name} ${def}\n`;
+  }
+};
+
+const getLabelAnimationName = () => {
+  const name = 'animation_label_opacity';
+  addKeyframe(name, '{ 0% { opacity: 0; } 99% { opacity: 0; } 100% { opacity: 1; } }');
+  return name;
+};
+
+const ITEM_HEIGHT = 48;
+
+const ExportBase = (props) => {
+  const [anchorEl, setAnchorEl] = useState(null);
+  const handleClick = (event) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const handleExport = option => () => {
+    const filter = node => (node.id !== 'iconButton');
+    const chart = document.body.childNodes[4];
```

Comment:
Let's obtain root element in a more firm way.
It can be obtained either by *id* (`document.querySelector`) or by *React.ref* (for now I don't know which way is better).

(1)
```jsx
<div id="widget-container">
  <Chart>
    ...
    <Export prop-to-get-root="#widget-container" />
  </Chart>
</div>
```

(2)
```jsx
<div ref={this.rootRef}>
  <Chart>
    ...
    <Export prop-to-get-root={this.rootRef} />
  </Chart>
</div>
```

Reply:
Done